concepts/decorators/about.md

```diff
@@ -1,2 +1,210 @@
-#TODO: Add about for this concept.
+# About
 
+Decorators are a feature of the Python language that can modify or register a function.
+
+## How to use them
+
+Decorators are placed just above a function in the code like so:
+
+```py
+@decorator
+def function():
+    pass
+```
+
+Some decorators take arguments:
+
+```py
+@decorator2(name="Bob")
+def function2():
+    pass
+```
+
+If a decorator takes any arguments, but you don't wish to pass any, you must still call it with parentheses for the decorator to work:
+
+```py
+@decorator3()
+def function3():
+    pass
+```
+
+Decorators are just syntactic sugar.
+An alternative way of doing the same as the above three examples is using higher-order functions:
+```py
+def function():
+    pass
+
+function = decorator(function)
+
+
+def function2():
+    pass
+
+function2 = decorator2(name="Bob")(function2)
+
+
+def function3():
+    pass
+
+function3 = decorator3()(function3)
+```
+
+## How to write them
+
+As said before, decorators can modify or register a function.
+The simplest decorator however, does absolutely nothing.
+
+Decorators are functions which return functions.
+They take one argument - the function which they have been added to.
+They also return one thing - the function which will be run by the program.
+This is usually either the function they were added to or a slightly modified version of that function, but it does not have to be - it can be any function whatsoever.
+
+The simplest decorator - one that does nothing - can be written as follows:
+```py
+def do_nothing(function):
+    return function
+```
+```py
+def function4a():
+    return 4
+
+@do_nothing
+def function4b():
+    return 4
+```
+```py
+>>>function4a()
+4
+>>>function4b()
+4
+```
+As you can see, both functions do the same thing, and when the decorator is added nothing changes.
+
+It is common to use the short form, `f`, instead of `function` which we will be using in the rest of the examples.
+
+A decorator that registers a function could be written as follows:
+```py
+functions = []
+
+def register_function(f):
+    functions.append(f)
+    return f
+```
+```py
+>>>functions
+[]
+```
+```py
+@register_function
+def function5():
+    return 5
+```
+```py
+>>>functions
+[<function function5 at 0x000001CA840AA700>]
+>>>function5()
+5
+```

```

Comment:
I like showing an example of "registering" ... but I think this could benefit from having a bit more detail.  Maybe something like what this snip from "Python Distilled" does?   Or another scenario where you're building a registry?

![image](https://user-images.githubusercontent.com/5923094/148607908-12a0514d-50be-4a20-b288-24b52c3f79b4.png)